### DISCUSSION TITLE
What key does pocketbase use to sign it's token?

### DISCUSSION THREAD

**Discussion by Irvingouj** [pocketbase/pocketbase#5227] [Q&A] [answered]
Is it randomly generated? can we specify what key we want to use?

**Comment by ganigeorgiev** [comment, marked as answer]
Without details what you are trying to do, the short answer would be No. Each token type (auth, verification, password reset, etc.) has a unique per user secret constructed from 2 parts:
- user unique part stored as `tokenKey` in each auth record
- shared part stored in the app settings (eg. `app.Settings().RecordAuthToken.Secret`)

_Please don't treat the discussions as llm chat and try to elaborate more on your problem to understand the broader picture of your use case (see also [XY problem](https://xyproblem.info/))._

**Reply by Irvingouj** [reply]
Thank you very much.
I'm trying to build an app where a thrid actor (a service runniing in local network) can validate the token of the user. I was wodering if it is possible to specify the public/private key pairs so the third actor would not need to make an extra request to validate the token.

**Reply by ganigeorgiev** [reply]
It is not possible. A call to the PocketBase server (ex. `authRefresh`) is necessary for the validation. For regular user auth tokens I'd argue that this is the safer way even if we were using a single fixed key because there is no guarantee that the issued token wasn't invalidated by the user (ex. when changing the password).
